src/crud_internal.c

```diff
@@ -119,188 +124,174 @@ int createObject( wrp_msg_t *reqMsg , wrp_msg_t **response)
 		}
 	}
 
-	if(reqMsg->u.crud.dest !=NULL)
+	if(jsonData !=NULL)
 	{
-		destVal = strdup(reqMsg->u.crud.dest);
-		ParodusInfo("destVal is %s\n", destVal);
-
-		if( (destVal != NULL))
-		{
-			child_ptr = strtok(destVal , "/");
+		free( jsonData );
+		jsonData = NULL;
+	}
 
-			/* Get the 1st object */
-			obj[0] = strdup( child_ptr );
-			ParodusPrint( "parent is %s\n", obj[0] );
+	if(reqMsg->u.crud.dest !=NULL)
+	{
+		ParodusInfo("reqMsg->u.crud.dest is %s\n", reqMsg->u.crud.dest);
+		child_ptr = strtok(reqMsg->u.crud.dest , "/");
 
-			free(destVal);
+		/* Get the 1st object */
+		obj[0] = strdup( child_ptr );
+		ParodusPrint( "parent is %s\n", obj[0] );
 
-			while( child_ptr != NULL)
+		while( child_ptr != NULL)
+		{
+			child_ptr = strtok( NULL, "/" );
+			if( child_ptr != NULL)
 			{
-				child_ptr = strtok( NULL, "/" );
-				if( child_ptr != NULL)
-				{
-					obj[i] = strdup( child_ptr );
-					ParodusPrint( "child obj[%d]:%s\n", i, obj[i] );
-					i++;
-				}
+				obj[i] = strdup( child_ptr );
+				ParodusPrint( "child obj[%d]:%s\n", i, obj[i] );
+				i++;
 			}
+		}
 
-			objlevel = i;
-			ParodusInfo( "Number of object level %d\n", objlevel );
+		objlevel = i;
+		ParodusInfo( "Number of object level %d\n", objlevel );
 
-			/* Valid request will be mac:14cfexxxx/parodus/tags/${name} which is objlevel 4 */
-			if(objlevel == 4)
+		/* Valid request will be mac:14cfexxxx/parodus/tags/${name} which is objlevel 4 */
+		if(objlevel == 4)
+		{
+			if(reqMsg->u.crud.payload != NULL)
 			{
-				if(reqMsg->u.crud.payload != NULL)
+				ParodusInfo("reqMsg->u.crud.payload is %s\n", (char *)reqMsg->u.crud.payload);
+				jsonPayload = cJSON_Parse( reqMsg->u.crud.payload );
+				if(jsonPayload !=NULL)
 				{
-					ParodusInfo("reqMsg->u.crud.payload is %s\n", (char *)reqMsg->u.crud.payload);
-
-					jsonPayload = cJSON_Parse( reqMsg->u.crud.payload );
-					if(jsonPayload !=NULL)
-					{
-						jsonPayloadSize = cJSON_GetArraySize( jsonPayload );
-						ParodusPrint( "jsonPayloadSize is %d\n", jsonPayloadSize );
-					}
-
-					if(jsonPayloadSize)
+					jsonPayloadSize = cJSON_GetArraySize( jsonPayload );
+					ParodusPrint( "jsonPayloadSize is %d\n", jsonPayloadSize );
+				}
+				if(jsonPayloadSize)
```

Comment:
Should be in the `if(jsonPayload !=NULL)` block?

Reply:
Incorporated this comment.